traversal/procedure/ProcedureEdge.java

```diff
@@ -103,6 +104,33 @@ private ProcedureEdge(VERTEX_FROM from, VERTEX_TO to, int order, Encoding.Direct
         }
     }
 
+    public static ProcedureEdge<?, ?> of(ProcedureVertex<?, ?> from, ProcedureVertex<?, ?> to,
+                                         StructureEdge<?, ?> structureEdge, int order, boolean isForward) {
+        Encoding.Direction.Edge dir = isForward ? FORWARD : BACKWARD;
+        if (structureEdge.isEqual()) {
+            return new Equal(from, to, order, dir);
+        } else if (structureEdge.isPredicate()) {
+            return new Predicate(from.asThing(), to.asThing(), order, dir, structureEdge.asPredicate().predicate());
+        } else if (structureEdge.isNative()) {
+            return Native.of(from, to, structureEdge.asNative(), order, isForward);
+        } else {
+            throw TypeDBException.of(UNRECOGNISED_VALUE);
+        }
+    }
```

Comment:
we now allow creating `ProcedureEdge` directly from `PlannerEdge` OR `StructureEdge`!  
This is pretty useful when we don't want to plan, but do something else to generate procedures (such as for `CombinationProcedure` or `VertexProcedure`)